wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -60,27 +59,152 @@ import           Universum
+
+instance Bounded a => Bounded (V1 a) where
+    minBound = V1 $ minBound @a
+    maxBound = V1 $ maxBound @a
+
+--
+-- Benign instances
+--
+
+instance ByteArray.ByteArrayAccess a => ByteArray.ByteArrayAccess (V1 a) where
+   length (V1 a) = ByteArray.length a
+   withByteArray (V1 a) callback = ByteArray.withByteArray a callback
+
+-- TODO(adinapoli) Rewrite it properly under CSL-2048.
+instance Arbitrary (V1 BackupPhrase) where
+    arbitrary = pure . V1 . BackupPhrase $ [
+          "shell"
+        , "also"
+        , "throw"
+        , "ramp"
+        , "grape"
+        , "chest"
+        , "setup"
+        , "mandate"
+        , "spare"
+        , "verb"
+        , "lemon"
+        , "test"
+        ]
+
+instance ToJSON (V1 BackupPhrase) where
+    toJSON (V1 (BackupPhrase wrds)) = toJSON wrds
+
+instance FromJSON (V1 BackupPhrase) where
+    parseJSON (Array wrds) = V1 . BackupPhrase . toList <$> traverse parseJSON wrds
+    parseJSON x            = typeMismatch "parseJSON failed for BackupPhrase" x
+
+mkPassPhrase :: Text -> Either Text Core.PassPhrase
+mkPassPhrase text =
+    case Base16.decode text of
+        Left e -> Left e
+        Right bs -> do
+            let bl = BS.length bs
+            -- Currently passphrase may be either 32-byte long or empty (for
+            -- unencrypted keys).
+            if bl == 0 || bl == Core.passphraseLength
+                then Right $ ByteArray.convert bs
+                else Left $ sformat
+                     ("Expected spending password to be of either length 0 or "%int%", not "%int)
+                     Core.passphraseLength bl
+
+instance ToJSON (V1 Core.PassPhrase) where
+    toJSON = String . Base16.encode . ByteArray.convert
+
+instance FromJSON (V1 Core.PassPhrase) where
+    parseJSON (String pp) = case mkPassPhrase pp of
+        Left e    -> fail (toString e)
+        Right pp' -> pure (V1 pp')
+    parseJSON x           = typeMismatch "parseJSON failed for PassPhrase" x
+
+instance Arbitrary (V1 Core.PassPhrase) where
+    arbitrary = fmap V1 arbitrary
+
+instance ToJSON (V1 Core.Coin) where
+    toJSON (V1 c) = toJSON . Core.unsafeGetCoin $ c
+
+instance FromJSON (V1 Core.Coin) where
+    parseJSON v = V1 . Core.mkCoin <$> parseJSON v
+
+instance Arbitrary (V1 Core.Coin) where
+    arbitrary = fmap V1 arbitrary
+
+instance ToJSON (V1 Core.Address) where
+    toJSON (V1 c) = String $ sformat addressF c
+
+instance FromJSON (V1 Core.Address) where
+    parseJSON (String a) = case Core.decodeTextAddress a of
+        Left e     -> fail $ "Not a valid Cardano Address: " <> toString e
+        Right addr -> pure (V1 addr)
+    parseJSON x = typeMismatch "parseJSON failed for Address" x
+
+instance Arbitrary (V1 Core.Address) where
+    arbitrary = fmap V1 arbitrary
+
+instance FromHttpApiData (V1 Core.Address) where
+    parseQueryParam = fmap (fmap V1) Core.decodeTextAddress
+
+instance ToHttpApiData (V1 Core.Address) where
+    toQueryParam (V1 a) = sformat build a
```

Comment:
You could have used `addressF` instead of `build` here as it's done few lines above (in `instance ToJSON (V1 Core.Address) where`) for consistency.
`addressF` is `build` with more concrete type.
Anyway, it's not important at all :)

Reply:
Ha, thanks for the tip! Lazyness, developer lazyness it was :) Will amend in subsequent PRs.